metamer/ftest/src/test/java/org/richfaces/tests/metamer/ftest/extension/creaper/ApplyPatch.java

```diff
@@ -0,0 +1,185 @@
+import org.wildfly.extras.creaper.core.offline.OfflineCommandContext;
+import org.wildfly.extras.creaper.core.online.CliException;
+import org.wildfly.extras.creaper.core.online.OnlineCommand;
+import org.wildfly.extras.creaper.core.online.OnlineCommandContext;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Command for patch application. OfflineCommand only for standalone mode!
+ */
+public final class ApplyPatch implements OnlineCommand, OfflineCommand {
+    private final String patchPath;
+    private final Boolean overrideAll;
+    private final Boolean overrideModules;
+    private final List<String> overridePaths;
+    private final List<String> preservePaths;
+
+    private ApplyPatch(Builder builder) {
+        this.patchPath = builder.patchPath;
+        this.overrideAll = builder.overrideAll;
+        this.overrideModules = builder.overrideModules;
+        this.overridePaths = builder.overridePaths;
+        this.preservePaths = builder.preservePaths;
+    }
+
+    @Override
+    public void apply(OnlineCommandContext ctx) throws CliException, CommandFailedException, IOException {
+        StringBuilder cmd = new StringBuilder("patch apply " + patchPath);
+        if (ctx.options.isDomain) {
+            cmd.append(" --host=");
+            cmd.append(ctx.options.defaultHost);
+        }
+        if (overrideAll != null && overrideAll) {
+            cmd.append(" --override-all");
+        }
+        if (overrideModules != null && overrideModules) {
+            cmd.append(" --override-modules");
+        }
+        if (overridePaths != null && !overridePaths.isEmpty()) {
+            cmd.append(" --override=");
+            cmd.append(PatchingConversions.flatten(overridePaths));
+        }
+        if (preservePaths != null && !preservePaths.isEmpty()) {
+            cmd.append(" --preserve=");
+            cmd.append(PatchingConversions.flatten(preservePaths));
+        }
+        ctx.client.executeCli(cmd.toString());
+    }
+
+    @Override
+    public String toString() {
+        return "ApplyPatch " + patchPath;
+    }
+
+    @Override
+    public void apply(OfflineCommandContext ctx) throws Exception {
+        StringBuilder cmd = new StringBuilder("patch apply " + patchPath);
+        if (overrideAll != null && overrideAll) {
+            cmd.append(" --override-all");
+        }
+        if (overrideModules != null && overrideModules) {
+            cmd.append(" --override-modules");
+        }
+        if (overridePaths != null && !overridePaths.isEmpty()) {
+            cmd.append(" --override=");
+            cmd.append(PatchingConversions.flatten(overridePaths));
+        }
+        if (preservePaths != null && !preservePaths.isEmpty()) {
+            cmd.append(" --preserve=");
+            cmd.append(PatchingConversions.flatten(preservePaths));
+        }
+        executeCliOffline(cmd.toString(), ctx);
```

Comment:
It is better to reuse the cmd command creation which is completely the same as for online command.